error: `if` expression missing an else clause
 --> tests/ui/auto_enum/compile-fail.rs:9:12
  |
9 |     } else if x > 3 {
  |            ^^

error: `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in this statement
  --> tests/ui/auto_enum/compile-fail.rs:18:5
   |
18 | /     let iter = match x {
19 | |         //~^ ERROR `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in...
20 | |         _ if y < 0 => return y..=0,
21 | |         _ => 2..=10,
22 | |     };
   | |______^

error: `#[auto_enum]` is required two or more branches or marker macros in total, there is no branch or marker macro in this statement
  --> tests/ui/auto_enum/compile-fail.rs:33:5
   |
33 | /     let iter = match x {
34 | |         //~^ ERROR `#[auto_enum]` is required two or more branches or marker macros in total, there is no branch or marker macro in this ...
35 | |         _ if y < 0 => return y..=0,
36 | |         _ => return 2..=10,
37 | |     };
   | |______^

error: `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in this statement
  --> tests/ui/auto_enum/compile-fail.rs:53:13
   |
53 | / ...   let _iter = match x {
54 | | ...       //~^ ERROR `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro ...
55 | | ...       _ if y < 0 => return y..=0,
56 | | ...       _ => 2..=10,
57 | | ...   };
   | |________^

error: `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in this statement
  --> tests/ui/auto_enum/compile-fail.rs:48:5
   |
48 | /     let iter = match x {
49 | |         //~^ ERROR `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in...
50 | |         _ if y < 0 => return y..=0,
51 | |         _ => {
...  |
59 | |         }
60 | |     };
   | |______^
